bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -274,7 +274,8 @@ static boolean isConvertibleToFloatingPointTypes(Object value) {
         switch (targetTypeTag) {
             case TypeTags.UNION_TAG:
                 for (Type memType : ((BUnionType) targetType).getMemberTypes()) {
-                    if (TypeChecker.getType(inputValue) == memType) {
+                    Type inputValueType = TypeChecker.getType(inputValue);
+                    if (inputValueType == memType || checkIntegerSubtypeMembers(inputValueType, memType)) {
```

Comment:
It seems currently with this fix it gives incorrect behaviour for some cases. For example
```bal
public function main() {
    int[] z = [257, 257, 257];
    (byte|int:Unsigned16)[] l = checkpanic z.cloneWithType();
}
```
previously succeeded in conversion as it should be. Now it fails giving the below result.
```logtalk
Running executable

error: {ballerina}NumberConversionError {"message":"'int' value '257' cannot be converted to 'byte'"}
        at ballerina.lang.value.0:cloneWithType(value.bal:86)
           temp:main(temp.bal:5)
```

Could you please add more tests covering different ranges in int subtypes.

Reply:
Resolved with https://github.com/ballerina-platform/ballerina-lang/pull/34136/commits/980da37aaf38e20954dceb368edf867887ae3ce1